src/laserMapping.cpp

```diff
@@ -542,8 +543,19 @@ void publish_map(rclcpp::Publisher<sensor_msgs::msg::PointCloud2>::SharedPtr pub
     }
     *pcl_wait_pub += *laserCloudWorld;
 
+    // Apply additional voxel filter to downsample the map before publishing
     sensor_msgs::msg::PointCloud2 laserCloudmsg;
-    pcl::toROSMsg(*pcl_wait_pub, laserCloudmsg);
+    if (mapPubVoxelFilter.getLeafSize().x() != 0.0f)
```

Comment:
Avoid filtering with possible invalid parameters.
```suggestion
    if (mapPubVoxelFilter.getLeafSize().x() > 0.0f)
```